fix(gui): Update GUI to support new API interfaces

## Summary

The GUI code (`internal/gui/`) fails to build due to interface mismatches with recent API changes.

## Build Errors

```
# github.com/mpyw/suve/internal/gui
internal/gui/param.go:93:35: cannot use client (variable of interface type ParamClient) as param.ListClient value in struct literal: ParamClient does not implement param.ListClient (missing method GetParameters)
internal/gui/staging.go:340:13: cannot use parser (variable of interface type "github.com/mpyw/suve/internal/staging".Parser) as "github.com/mpyw/suve/internal/staging".EditStrategy value in struct literal: "github.com/mpyw/suve/internal/staging".Parser does not implement "github.com/mpyw/suve/internal/staging".EditStrategy (missing method FetchCurrentValue)
```

## Root Cause

1. **`ParamClient` missing `GetParametersAPI`**: The `param.ListClient` interface now requires `paramapi.GetParametersAPI` (added in #53 for batch parameter fetching), but the GUI's `ParamClient` interface doesn't include it.

2. **`Parser` missing `FetchCurrentValue`**: The staging `EditStrategy` interface requires a method that `Parser` doesn't implement.

## Reproduction

```bash
go build -tags dev ./internal/gui/...
```

## Notes

- The GUI has build tags `//go:build production || dev`, so these errors don't appear in regular `make test` or `make lint`
- The CLI functionality is unaffected

--- a/internal/gui/app.go
+++ b/internal/gui/app.go
@@ -21,6 +21,7 @@ type ParamClient interface {
 	staging.ParamClient
 	// For usecases (additional methods not in staging.ParamClient)
 	paramapi.DescribeParametersAPI
+	paramapi.GetParametersAPI
 	paramapi.ListTagsForResourceAPI
 }
 

--- a/internal/gui/staging.go
+++ b/internal/gui/staging.go
@@ -331,13 +331,13 @@ func (a *App) StagingAdd(service, name, value string) (*StagingAddResult, error)
 		return nil, err
 	}
 
-	parser, err := a.getParser(service)
+	strategy, err := a.getEditStrategy(service)
 	if err != nil {
 		return nil, err
 	}
 
 	uc := &stagingusecase.AddUseCase{
-		Strategy: parser,
+		Strategy: strategy,
 		Store:    store,
 	}
 	result, err := uc.Execute(a.ctx, stagingusecase.AddInput{

